### DISCUSSION TITLE
First request fails consistently.

### DISCUSSION THREAD

**Discussion by FredLackeyOfficial** [prisma/prisma#15636] [Q&A]
### Bug description

The first request to a Postgres database causes an exception to be thrown.  Calling the same code in the same Node service works perfectly from that point forward.  This is a Node service running within Docker on a 32GB M1 MacBook Pro.  The exception basics are:

```
Invalid `prisma.widget.findMany()` invocation in...

Can't reach database server at `postgres`:`5432`

Please make sure your database server is running at `postgres`:`5432`.
```

Have not looked into it, but it almost feels like the PrismaClient is using some type of singleton pattern and throwing the error before some underlying piece initializing. That could explain why subsequent calls succeed.

### How to reproduce

Use the following code in an Express route:

```
const { PrismaClient } = require('@prisma/client');

const prisma = new PrismaClient();

const getAll = async (req, res, next) => {

  const { uid } = req.body;
  const existing = await prisma.widget.findMany({
    where: {
      uid : uid.trim().toUpperCase(),
    }
  });

  return res.json(existing);
}

module.exports = getAll;
```

### Expected behavior

Return the collection of matching items without throwing an error.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model widget {
  id      Int       @id(map: "pk_widget") @default(autoincrement()) @db.SmallInt
  uid     String    @db.Char(32)
  ac      DateTime  @map("_ac") @db.Timestamp(6)
}
```

### Environment & setup

macos 12.6
docker 20.10.17, build 100c701
docker compose 2.10.2
node 16.17.0
express 4.18.1

### Prisma Version

```
@prisma/client : 4.4.0
prisma 4.4.0
```

**Comment by austincrim** [comment]
Hey @FredLackeyOfficial 👋,

If you could provide a minimal reproduction repository with your relevant setup, I would love to dive into this one.

**Reply by vladi-stevanovic** [reply]
Hello @FredLackeyOfficial did you perhaps have a chance to look into creating a minimal reproduction for this? We'd love to continue troubleshooting the problem and get to a solution!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
